openapi.yaml

```diff
@@ -148,38 +166,56 @@ components:
         via_waypoints:
           type: array
           nullable: true
-          default: null
-          description: A Mapbox and Valhalla extension which indicates which waypoints are passed through rather than creating a new leg.
+          description: >-
+            Indicates which waypoints are passed through rather than creating a new leg.
+            This is a Valhalla extension to the OSRM spec.
           items:
             $ref: '#/components/schemas/ViaWaypoint'
+        # TODO: admins
     RouteStep:
       type: object
+      description: >-
+        A maneuver such as a turn or merge, followed by travel along a single road or path.
+      required:
+        - distance
+        - duration
+        - geometry
+        - mode
+        - maneuver
       properties:
         distance:
           type: number
           format: double
-          description: The distance traveled by the route, in double meters.
+          description: The distance traveled by the route, in meters.
         duration:
           type: number
           format: double
-          description: The estimated travel time, in double number of seconds.
+          description: The estimated travel time, in number of seconds.
         geometry:
-          type: object
+          type: object  # FIXME
```

Comment:
This is going to be an encoded polyline in all of our use cases, but technically OSRM supports GeoJSON too. Any objection to just skipping that? It's going to be absurdly inefficient 😂